frontend_server/lib/server.dart

```diff
@@ -259,6 +259,23 @@ Future<int> starter(
   }
 
   if (options['train']) {
+    String sdkRoot = options['sdk-root'];
+    if (sdkRoot.startsWith("//out/") && sdkRoot.endsWith("/gen")) {
+      // This becomes '../../out/whatnot/flutter_patched_sdk'
+      sdkRoot = '../..${sdkRoot.substring(1, sdkRoot.length-4)}/flutter_patched_sdk';
```

Comment:
This is really hacky, I don't like it. Is there no way to make GN properly rebase paths and get absolute path in here?

Maybe @zanderso knows the trick?

Reply:
How that you implicitly told me what this rebase_path command I've seen used does, then yes. Fixed.